pkg/kubelet/kubelet.go

```diff
@@ -2700,14 +2700,16 @@ func (kl *Kubelet) handlePodResourcesResize(pod *v1.Pod) {
 		if err := kl.statusManager.SetPodAllocation(updatedPod); err != nil {
 			//TODO(vinaykul,InPlacePodVerticalScaling): Can we recover from this in some way? Investigate
 			klog.ErrorS(err, "SetPodAllocation failed", "pod", klog.KObj(pod))
+			return
 		}
-		*pod = *updatedPod
+		pod = updatedPod
 	}
 	if resizeStatus != "" {
 		// Save resize decision to checkpoint
 		if err := kl.statusManager.SetPodResizeStatus(pod.UID, resizeStatus); err != nil {
 			//TODO(vinaykul,InPlacePodVerticalScaling): Can we recover from this in some way? Investigate
 			klog.ErrorS(err, "SetPodResizeStatus failed", "pod", klog.KObj(pod))
+			return
 		}
 		pod.Status.Resize = resizeStatus
```

Comment:
Looking at this comment: https://github.com/kubernetes/kubernetes/pull/116706#discussion_r1140217546

`pod` is now a reference to `updatedPod` so this operates on the `pod`'s copy. Should not cause a race

Reply:
maybe from code cleanless perspective keep the `updatedPod`instead of `pod` being used all the way to the `kl.podManager.UpdatePod(pod)`.

Reply:
is fit ever false when resizeStatus is non-empty? if so, we'll miss the `pod = updatedPod` assignment and this will still mutate the original pod here, right?

Reply:
it looks like there are return paths from canResizePod that would lead us to mutate the original pod here:

```
return false, nil, v1.PodResizeStatusInfeasible
...
return false, nil, v1.PodResizeStatusDeferred
```

Reply:
Fixed. PTAL, thankx